src/fidesops/task/graph_task.py

```diff
@@ -160,11 +162,16 @@ def to_dask_input_data(self, *data: List[Row]) -> Dict[str, List[Any]]:
                 Tuple[FieldPath, FieldPath]
             ] = self.incoming_field_path_map[collection_address]
 
+            logger.info(
+                f"Consolidating incoming data into {self.traversal_node.node.address} from {collection_address}."
+            )
             for row in rowset:
                 for foreign_field_path, local_field_path in field_mappings:
-                    new_value = foreign_field_path.retrieve_from(row)
-                    if new_value:
-                        append(output, local_field_path.string_path, new_value)
+                    new_values: List = consolidate_query_matches(
+                        row=row, target_path=foreign_field_path
+                    )
+                    if new_values:
+                        append(output, local_field_path.string_path, new_values)
```

Comment:
This is the final primary change. Before, we were using `foreign_field_path.retrieve_from(row)` to get the data from the record and pass into the next collection. However, array data doesn't necessarily have a direct path, and array data can have *multiple* matched paths. So we get the multiple matches, and consolidate them into a single list.